storage/app/src/main/java/com/google/firebase/referencecode/storage/KotlinStorageActivity.kt

```diff
@@ -0,0 +1,405 @@
+package com.google.firebase.referencecode.storage
+
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.net.Uri
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.widget.ImageView
+import com.google.android.gms.tasks.Continuation
+import com.google.android.gms.tasks.Task
+import com.google.firebase.referencecode.storage.interfaces.StorageActivityInterface
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageMetadata
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.UploadTask
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileInputStream
+
+class KotlinStorageActivity : AppCompatActivity(), StorageActivityInterface {
+    // [START storage_field_declaration]
+    // [END storage_field_declaration]
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_storage)
+
+        // [START storage_field_initialization]
+        val storage = FirebaseStorage.getInstance()
+        // [END storage_field_initialization]
+
+        includesForCreateReference()
+    }
+
+    override fun includesForCreateReference() {
+        val storage = FirebaseStorage.getInstance()
+
+        // ## Create a Reference
+
+        // [START create_storage_reference]
+        // Create a storage reference from our app
+        var storageRef = storage.reference
+        // [END create_storage_reference]
+
+        // [START create_child_reference]
+        // Create a child reference
+        // imagesRef now points to "images"
+        var imagesRef: StorageReference? = storageRef.child("images")
```

Comment:
Does `.child()` return something `Nullable`?  If so I will report that internally as a bug.

Reply:
It does not. But `imagesRef` is later reassigned to `spaceRef.parent`. The `parent` property does return a Nullable `StorageReference`

Reply:
Hmm ok, we can leave it this way for now then.